src/plugins/workspace/server/utils.ts

```diff
@@ -113,3 +115,25 @@ export const getDataSourcesList = (client: SavedObjectsClientContract, workspace
       }
     });
 };
+
+export const checkAndSetDefaultDataSource = async (
+  uiSettingsClient: IUiSettingsClient,
+  dataSources: string[],
+  isNeededCheck: boolean
+) => {
+  if (dataSources?.length > 0) {
+    if (!isNeededCheck) {
+      // Create# Will set first data source as default data source.
+      await uiSettingsClient.set(DEFAULT_DATA_SOURCE_UI_SETTINGS_ID, dataSources[0]);
+    } else {
```

Comment:
What if we always check if the default data source is set?

Reply:
I would be uninclined to do this, we don't need to send an additional request to check default data source for a new workspace even if this could make code less.